src/components/AddItem.js

```diff
@@ -24,9 +27,27 @@ const AddItem = ({ token }) => {
         });
   };
 
-  const handleSubmit = (event) => {
+  const handleSubmit = async (event) => {
     event.preventDefault();
-    db.collection(token).add(formData);
+    const itemNameInput = formData.itemName;
+
+    // check to see if the itemName already exists in the user's list - convert input to lowercase and remove any punctuation & white space
+    const cleanInput = itemNameInput.toLowerCase().replace(/[\W_]*|/g, '');
+
+    // query the collection and filter for match - if match exists send the user an error message "this item already exists in your list..." & don't allow them to submit it
+    const queryCollection = db.collection(token);
+    const snapshot = await queryCollection
+      .where('cleanedUpItemName', '==', cleanInput)
+      .get();
+    if (!snapshot.empty) {
+      setHidden(true);
+      return;
+    }
+    // else add the user's inputted item to the database
+    db.collection(token).add({
+      ...formData,
+      cleanedUpItemName: cleanInput,
```

Comment:
Nit - One alternative I had mentioned in Slack and idea for a future refactor (there are always alternatives and that's what makes things fun!) would be to make the `cleanInput` be the document's id rather than a property on the document itself. That could make retrieval of the document a little cleaner. If you did that, you may consider renaming `cleanInput` to something like `itemId`.
This 100% works as is and I think either way works great.